types/mathjs/index.d.ts

```diff
@@ -1,2252 +1,4568 @@
+         * must be a lower triangular matrix.
+         * @param L A N x N matrix or array (L)
+         * @param b A column vector with the b values
+         * @returns A column vector with the linear system solution (x)
+         */
+        lsolve(
+            L: Matrix | MathArray,
+            b: Matrix | MathArray
+        ): Matrix | MathArray;
+
+        /**
+         * Calculate the Matrix LU decomposition with partial pivoting. Matrix A
+         * is decomposed in two matrices (L, U) and a row permutation vector p
+         * where A[p,:] = L * U
+         * @param A A two dimensional matrix or array for which to get the LUP
+         * decomposition.
+         * @returns The lower triangular matrix, the upper triangular matrix and
+         * the permutation matrix.
+         */
+        lup(
+            A?: Matrix | MathArray
+        ): { L: MathArray | Matrix; U: MathArray | Matrix; P: number[] };
+
+        /**
+         * Solves the linear system A * x = b where A is an [n x n] matrix and b
+         * is a [n] column vector.
+         * @param A Invertible Matrix or the Matrix LU decomposition
+         * @param b Column Vector
+         * @param order The Symbolic Ordering and Analysis order, see slu for
+         * details. Matrix must be a SparseMatrix
+         * @param threshold Partial pivoting threshold (1 for partial pivoting),
+         * see slu for details. Matrix must be a SparseMatrix.
+         * @returns Column vector with the solution to the linear system A * x =
+         * b
+         */
+        lusolve(
+            A: Matrix | MathArray | number,
+            b: Matrix | MathArray,
+            order?: number,
+            threshold?: number
+        ): Matrix | MathArray;
+
+        /**
+         * Calculate the Matrix QR decomposition. Matrix A is decomposed in two
+         * matrices (Q, R) where Q is an orthogonal matrix and R is an upper
+         * triangular matrix.
+         * @param A A two dimensional matrix or array for which to get the QR
+         * decomposition.
+         * @returns Q: the orthogonal matrix and R: the upper triangular matrix
+         */
+        qr(
+            A: Matrix | MathArray
+        ): { Q: MathArray | Matrix; R: MathArray | Matrix };
+
+        /**
+         * Transform a rationalizable expression in a rational fraction. If
+         * rational fraction is one variable polynomial then converts the
+         * numerator and denominator in canonical form, with decreasing
+         * exponents, returning the coefficients of numerator.
+         * @param expr The expression to check if is a polynomial expression
+         * @param optional scope of expression or true for already evaluated
+         * rational expression at input
+         * @param detailed  optional True if return an object, false if return
+         * expression node (default)
+         * @returns The rational polynomial of expr
+         */
+        rationalize(
```

Comment:
Looks like the return type here can be more specific:
```typescript
rationalize(expr: MathNode | string, optional?: object | boolean, detailed: true): { expression: MathNode | string, variables: string[], coefficients: MathType[] };
rationalize(expr: MathNode | string, optional?: object | boolean, detailed: false): MathNode;
```
Also I believe the last two arguments should be optional.